### PR TITLE
NOISSUE: alternative testrail output

### PR DESCRIPTION
**Pull request by bigbes**
output can be like that:
```UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C1 Status=PASS message= writeDuration="97.0µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C2 Status=PASS message= writeDuration="11.0µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C3 Status=PASS message= writeDuration="16.0µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C4 Status=SKIP SkippedLink="jira link 1" message= writeDuration="5.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C5 Status=SKIP SkippedLink="jira link 2" message= writeDuration="4.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C6 Status=SKIP SkippedLink="jira link 3" message= writeDuration="18.0µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C7 Status=FAIL message= writeDuration="8.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C8 Status=FAIL message= writeDuration="8.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C9 Status=FAIL message= writeDuration="6.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C10 Status=PASS message= writeDuration="11.0µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C11 Status=FAIL message= writeDuration="8.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C12 Status=FAIL message= writeDuration="6.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C13 Status=FAIL message= writeDuration="8.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C14 Status=SKIP SkippedLink="jira link 4" message= writeDuration="5.00µs"
UNK time=2020-07-21T15:21:34+03:00 caller=/data/go/src/github.com/insolar/assured-ledger/ledger-core/testutils/investigation/testrail.go:82 loginstance=global_default Case=C15 Status=FAIL message= writeDuration="9.00µs"
```

or like that:
```TestHappyPath: testrail.go:88: C1 PASS
    TestHappyPathParallel/parallel_1: testrail.go:88: C2 PASS
    TestHappyPathParallel/parallel_2: testrail.go:88: C3 PASS
    TestSkipped: testrail.go:88: C4 SKIP "jira link 1"
    TestParallelSkipped/parallel_1: testrail.go:88: C5 SKIP "jira link 2"
    TestParallelSkipped/parallel_2: testrail.go:88: C6 SKIP "jira link 3"
    TestFail: testrail.go:88: C7 FAIL
    TestFailNow: testrail.go:88: C8 FAIL
    TestParallelFail1/parallel_1: testrail.go:88: C9 FAIL
    TestParallelFail1/parallel_2: testrail.go:88: C10 PASS
    TestParallelFail2/parallel_1: testrail.go:88: C11 FAIL
    TestParallelFail2/parallel_2: testrail.go:88: C12 FAIL
    TestParallelSkipAndFail/parallel_1: testrail.go:88: C13 FAIL
    TestParallelSkipAndFail/parallel_2: testrail.go:88: C14 SKIP "jira link 4"
    TestPanics: testrail.go:88: C15 FAIL
```

It looks like enough for testrail cases, first one (with global logger) is not initialized right now

### CODE REVIEW

File: ledger-core/virtual/integration/delegated_finished_test.go [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-	return insolar.MustSerialize(foundation.Result{
-		Error:   nil,
-		Returns: []interface{}{ret0},
-	})
-}
-
-type callMethodFunc = func(ctx context.Context, callContext *call.LogicContext, code reference.Global, data []byte, method string, args []byte) (newObjectState []byte, methodResults []byte, err error)
-type callConstructorFunc = func(ctx context.Context, callContext *call.LogicContext, code reference.Global, name string, args []byte) (objectState []byte, result []byte, err error)
-
-func mockExecutor(t *testing.T, server *utils.Server, callMethod callMethodFunc, constructorFunc callConstructorFunc) {
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/delegated_finished_test.go:37]: Why did you delete this file? By mistake?

Comment (bigbes) [reply ledger-core/virtual/integration/delegated_finished_test.go:37]: It was unused

File: ledger-core/testutils/investigation/testrail_test.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package investigation
+
+// import (
```
Comment (ElSacramento) [inline ledger-core/testutils/investigation/testrail_test.go:8]: Will be uncommented later?

Comment (bigbes) [reply ledger-core/testutils/investigation/testrail_test.go:8]: No, it wont. It's was for testing purposes. Go doesn't support expected fail tests and other :(
I can remove it, if you suggest

Comment (ElSacramento) [reply ledger-core/testutils/investigation/testrail_test.go:8]: Let's remove it

File: ledger-core/Makefile.testing [hunk 3]
```diff
@@ -16,7 +18,7 @@ test_unit: ## run all unit tests
 
 .PHONY: test_with_coverage
 test_with_coverage: ## run unit and slow tests with generation of coverage file
-	go-acc --covermode=count --output=coverage.tmp.out --ignore drafts,load ./... -- -v -tags="slowtest copylogtxt" -count=1 $(TEST_ARGS) -args TESTLOG_OUT=$(EXTENDED_LOG) TESTLOG_MARKS=1 TESTLOG_TXT=1 2>&1 | tee "integration_test.log"
+	go-acc --covermode=count --output=coverage.tmp.out --ignore drafts,load ./... -- -v -tags="slowtest copylogtxt" -count=1 $(TEST_ARGS) -args TESTLOG_OUT=$(EXTENDED_INTEGRATION_LOG) TESTLOG_MARKS=1 TESTLOG_TXT=1 2>&1 | tee "integration_test.log"
```
Comment (cyraxred) [inline ledger-core/Makefile.testing:21]: do we still use tee?

File: ledger-core/application/functest/main_func_test.go [hunk 4]
```diff
@@ -12,9 +12,19 @@ import (
...
 
+type testingLogger struct{}
+
+func (t testingLogger) Helper()                {}
+func (t testingLogger) Log(i ...interface{})   {}
+func (t testingLogger) Error(i ...interface{}) {}
+func (t testingLogger) Fatal(i ...interface{}) {}
+
 func TestMain(m *testing.M) {
+	instestlogger.SetTestOutput(testingLogger{})
```
Comment (cyraxred) [inline ledger-core/application/functest/main_func_test.go:26]: Please use SetTestOutputWithStub()

File: ledger-core/Makefile.testing (hunk 3 above)
Comment (bigbes) [reply ledger-core/Makefile.testing]: removed

File: ledger-core/application/functest/wallet_add_amount_test.go [hunk 5]
```diff
@@ -12,11 +12,14 @@ import (
...
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/insolar/assured-ledger/ledger-core/testutils/investigation"
 )
 
 // Creates wallet, calls /wallet/add_amount and checks it's response body.
 func TestWalletAddAmount(t *testing.T) {
-	t.Log("C4921")
+	investigation.LogCase(t, "C4921")
```
Comment (cyraxred) [inline ledger-core/application/functest/wallet_add_amount_test.go:21]: "investigation log case" doesn't read well
May be "testrail" or "insrail" or something else pls?

File: ledger-core/Makefile.testing [hunk 6]
```diff
@@ -16,7 +18,7 @@ test_unit: ## run all unit tests
 
 .PHONY: test_with_coverage
 test_with_coverage: ## run unit and slow tests with generation of coverage file
-	go-acc --covermode=count --output=coverage.tmp.out --ignore drafts,load ./... -- -v -tags="slowtest copylogtxt" -count=1 $(TEST_ARGS) -args TESTLOG_OUT=$(EXTENDED_LOG) TESTLOG_MARKS=1 TESTLOG_TXT=1 2>&1 | tee "integration_test.log"
+	go-acc --covermode=count --output=coverage.tmp.out --ignore drafts,load ./... -- -v -tags="slowtest copylogtxt" -count=1 $(TEST_ARGS) -args TESTLOG_OUT=$(EXTENDED_INTEGRATION_LOG) TESTLOG_MARKS=1 TESTLOG_TXT=1 2>&1
```
Comment (sexyhamster) [inline ledger-core/Makefile.testing:21]: I guess we don't need tee anymore

File: ledger-core/virtual/integration/contract_calls_test.go [hunk 7]
```diff
@@ -100,8 +101,9 @@ func Test_NoDeadLock_WhenOutgoingComeToSameNode(t *testing.T) {
 	for _, test := range table {
 		test := test
 		t.Run(test.name, func(t *testing.T) {
-			t.Log(test.testCaseID)
-			t.Skip("https://insolar.atlassian.net/browse/PLAT-432")
+			investigation.LogCase(t, test.testCaseID)
+			investigation.LogSkip(t, "https://insolar.atlassian.net/browse/PLAT-432")
```
Comment (cyraxred) [inline ledger-core/virtual/integration/contract_calls_test.go:105]: as these can't be separate, I'd suggest a form like 

investigation.Skip(t, "https://insolar.atlassian.net/browse/PLAT-432").Case(test.testCaseID)

File: ledger-core/application/functest/main_func_test.go (hunk 4 above)
Comment (sexyhamster) [inline ledger-core/application/functest/main_func_test.go:26]: Could you explain what is going on here? Do you want to suppress log preparation?

File: ledger-core/application/functest/wallet_add_amount_test.go (hunk 5 above)
Comment (sexyhamster) [inline ledger-core/application/functest/wallet_add_amount_test.go:21]: I'm not sure about naming investigation doesn't sound like the thing you want to put in every test scenario. Maybe something like testRailOutput?

File: ledger-core/application/functest/main_func_test.go (hunk 4 above)
Comment (cyraxred) [reply ledger-core/application/functest/main_func_test.go:26]: there is no testing.T available here

Comment (cyraxred) [reply ledger-core/application/functest/main_func_test.go:26]: had the same issue with network testing suites

File: ledger-core/Makefile.testing (hunk 6 above)
Comment (bigbes) [reply ledger-core/Makefile.testing:21]: already removed

File: ledger-core/application/functest/wallet_add_amount_test.go (hunk 5 above)
Comment (bigbes) [reply ledger-core/application/functest/wallet_add_amount_test.go]: fixed

File: ledger-core/application/functest/main_func_test.go (hunk 4 above)
Comment (bigbes) [reply ledger-core/application/functest/main_func_test.go]: fixed

File: ledger-core/virtual/integration/contract_calls_test.go (hunk 7 above)
Comment (bigbes) [reply ledger-core/virtual/integration/contract_calls_test.go]: fixed

File: ledger-core/application/functest/wallet_add_amount_test.go (hunk 5 above)
Comment (bigbes) [reply ledger-core/application/functest/wallet_add_amount_test.go]: fixed

File: ledger-core/application/functest/main_func_test.go (hunk 4 above)
Comment (bigbes) [reply ledger-core/application/functest/main_func_test.go]: fixed
